staking/staking-specification.md

```diff
@@ -0,0 +1,815 @@
+
+/// @dev Returns the stake delegated by a given owner.
+/// @param owner of stake.
+/// @return Delegated stake for owner.
+function getStakeDelegatedByOwner(address owner)
+    external
+    view
+    returns (IStructs.StakeBalance memory balance);
+
+/// @dev Returns the amount stake that can be withdrawn for a given owner.
+/// @param owner of stake.
+/// @return Withdrawable stake for owner.
+function getWithdrawableStake(address owner)
+    public
+    view
+    returns (uint256);
+
+/// @dev Returns the stake delegated to a specific staking pool, by a given owner.
+/// @param owner of stake.
+/// @param poolId Unique Id of pool.
+/// @return Stake delegaated to pool by owner.
+function getStakeDelegatedToPoolByOwner(address owner, bytes32 poolId)
+    public
+    view
+    returns (IStructs.StakeBalance memory balance);
+
+/// @dev Returns the total stake delegated to a specific staking pool,
+///      across all members.
+/// @param poolId Unique Id of pool.
+/// @return Total stake delegated to pool.
+function getTotalStakeDelegatedToPool(bytes32 poolId)
+    public
+    view
+    returns (IStructs.StakeBalance memory balance);
+```
+
+## 6 Liquidity Incentives
+
+Liquidity incentives are used to align market participants with long-term objectives of the 0x protocol. Fees are charged by the 0x protocol and paid to market makers as a reward for their trade volume and amount of ZRX staked.
+
+<p align="center"><img src="./assets/ProtocolFee.png" width="700" /></p>
+
+The protocol fee is paid in either WETH or ETH. If ETH is not included in the transaction (by setting `msg.value`) then the fee will be taken in WETH from the taker, as illustrated in the diagram above.
+
+Note also that the WETH Asset Proxy is distinct from the standard ERC20 Asset Proxy, used by the exchange. There are two reasons for this. Firstly, this separation means that users must explicitly opt-in to fees (no hidden fees). Secondly, the Staking Contract only needs access to WETH, whereas the ERC20 Proxy would provide access to _all_ tokens that a user has given allowance for; since the Staking Contract is upgradable, this separation reduces the attack surface.
+
+### 6.1 Market Making
+
+We want to align market makers with the long-term objectives of the 0x protocol; however, the immobility of staked ZRX exposes makers to potential short-term volatility in the crypto markets. This risk is mitigated through staking pools.
+
+A maker creates a pool, which can be delegated to by any other staker. When computing a maker's reward, we account for all the stake in their pool. We favor ZRX staked directly by the maker by assigning a lower weight (90%) to ZRX delegated by other stakers.
+
+Market makers incentivize delegators to join their pool by setting aside a fixed percentage of their reward for the members of their pool. A member receives an amount of this percentage that is proportional to how much stake they have delegated to the pool.
```

Comment:
Are you talking about `operatorShare` here, or just the normal reward disbursements?

Reply:
This is the operator share